sale_manual_delivery/wizard/manual_proc_view.xml

```diff
@@ -0,0 +1,64 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
+
+    <record id='manual_delivery_wizard_form' model='ir.ui.view'>
+        <field name="name">manual.delivery.wizard.form</field>
+        <field name="model">manual.delivery</field>
+        <field name='arch' type='xml'>
+            <form string="Create Manually Delivery" name="manual_procurment_form">
+                <sheet>
+                    <group name="date">
+                        <field name='date_planned' required="1"/>
+                    </group>
+                    <group name="carrier">
+                        <field name='carrier_id'/>
+                    </group>
+                    <group name="address">
+                        <field name="commercial_partner_id" invisible="1"/>
+                        <field name='partner_id'/>
+                    </group>
+                    <group name="route">
+                        <field name='route_id' groups="sale_stock.group_route_so_lines"/>
+                    </group>
+                    <field name='line_ids'>
+                        <tree editable="bottom" create="0">
+                            <field name='manual_delivery_id' invisible="1"/>
+                            <field name='order_line_id' invisible="1"/>
+                            <field name='product_id'/>
+                            <field name='line_description'/>
+                            <field name='ordered_qty'/>
+                            <field name='existing_qty'/>
+                            <field name='remaining_qty'/>
+                            <field name='to_ship_qty'/>
+                        </tree>
+                    </field>
+                </sheet>
+                <footer>
+                    <button name='record_picking'
+                            string='Create Picking'
+                            class='btn-primary'
+                            type='object'/>
+                    <button string="Cancel" class="btn-default" special="cancel" />
+                </footer>
+            </form>
+
+        </field>
+    </record>
+
+    <act_window id='action_wizard_manual_delivery'
+                name='Create Manual Delivery'
+                res_model='manual.delivery'
+                view_mode='form'
+                target='new'/>
+
+    <record model="ir.values" id="sale_order_line_make_invoice">
+      <field name="model_id" ref="sale.model_sale_order_line" />
+      <field name="name">Create Manual Delivery</field>
+      <field name="key2">client_action_multi</field>
+      <field name="value" eval="'ir.actions.act_window,' + str(ref('action_wizard_manual_delivery'))" />
```

Comment:
you can define two kinds of wizards one is generated from sale order and will have wider functionality to handle delivery per order like date, route, address, origin etc
the second one is initialized on sale order lines and in that case, it will have more strict functionality as it can get lines from different orders and partners  
in this way, we can satisfy the needs of all sides and we just need to define two forms of wizard and make some methods more generic to be able to customize them locally in case of need

Reply:
I think it is the same thing, the sale order is just a bunch of sale order lines, each one is delivered individually. The only thing that is common in a sale order line is the partner. The other fields, like route, delivery address, schedule date could be different within a sale order line. 

If we restrict the selection of the lines we will be removing the possibility of changing that information. The only restriction in my opinion should be the partner, so only lines from the same partner should be selected.